Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -4350,9 +4350,8 @@ Ref<MutableStyleProperties> ComputedStyleExtractor::copyPropertiesInSet(const CS
 Ref<MutableStyleProperties> ComputedStyleExtractor::copyProperties()
 {
     Vector<CSSProperty> list;
-    list.reserveInitialCapacity(firstShorthandProperty - firstCSSProperty);
-    for (unsigned i = firstCSSProperty; i < firstShorthandProperty; ++i) {
-        auto propertyID = convertToCSSPropertyID(i);
+    list.reserveInitialCapacity(numCSSPropertyLonghands);
+    for (auto propertyID : allLonghandCSSProperties()) {
```

Comment:
Not necessarily something to actually change, but what is your feeling on this:

```c++
auto list = allLonghandCSSProperties().compactMap([](auto propertyID) -> std::optional<CSSProperty> {
    if (auto value = propertyValue(propertyID))
        return CSSProperty(propertyID, WTFMove(value)));
    return std::nullopt;
});
```

(though, I think it would probably be `auto list = WTF::compactMap(allLonghandCSSProperties(), ...` since Vector doesn't have a `compactMap` of its own).

Reply:
I love that; I am tempted to follow up with it.